drift-transport-netty/src/test/java/com/facebook/drift/transport/netty/client/TestConnectionPool.java

```diff
@@ -60,10 +60,21 @@ public void testPooling()
         }
     }
 
+    @Test
+    public void testPoolingWithMultipleConnections()
+    {
+        try (ConnectionPool pool = new ConnectionPool(new TestingConnectionManager(), new DefaultEventLoopGroup(), 10, 2, new Duration(1, MINUTES))) {
+            HostAndPort address = HostAndPort.fromParts("localhost", 1234);
+
+            futureGet(pool.getConnection(PARAMETERS, address));
+            futureGet(pool.getConnection(PARAMETERS, address));
```

Comment:
How to add assertion to ensure that we get a different connection. Try a few times and then see if we get a diff connection?

Reply:
In my initial PR with round robin, test was written the way you suggested to verify a different connection is generated. After changing to random connection id,  this test was modified to only test it works.